data/systemd/yggdrasil-sysuser.conf.in

```diff
@@ -0,0 +1,4 @@
+#Type Name     ID   GECOS                   Home directory             Shell
+u     @user@   -    "yggdrasil system user" @sharedstatedir@/yggdrasil -
+g     @group@  -
+m     @user@   @group@
```

Comment:
The output of `packit build local` says: `ignoring unsupported sysuser type: m` despite installed version of systemd should support it according documentation: https://www.freedesktop.org/software/systemd/man/latest/sysusers.d.html

Or is it bug in some rpm macro? Like this: https://github.com/rpm-software-management/rpm/blob/master/macros.in#L1341 Weird.

Reply:
The builds in COPR that packit triggered don't seem to have that same warning line. I couldn't find it in [RHEL 8](https://download.copr.fedorainfracloud.org/results/packit/RedHatInsights-yggdrasil-224/rhel-8-x86_64/07321759-yggdrasil/builder-live.log.gz)

Reply:
I just tried building locally with `packit build in-mock`, and the RPMs built fine.

Reply:
I was able to build the RPM after restarting of system. When I installed the RPM to the system, and I tried to start the service, then start of `yggdrasil.service` was terminated with following error:

```
root@thinkpad-p1:/home/jhnidek/github/redhat-insights/yggdrasil# systemctl status yggdrasil.service 
○ yggdrasil.service - yggdrasil system service
     Loaded: loaded (/usr/lib/systemd/system/yggdrasil.service; disabled; preset: disabled)
    Drop-In: /usr/lib/systemd/system/service.d
             └─10-timeout-abort.conf
     Active: inactive (dead)
       Docs: https://github.com/RedHatInsights/yggdrasil
root@thinkpad-p1:/home/jhnidek/github/redhat-insights/yggdrasil# systemctl start yggdrasil.service 
Job for yggdrasil.service failed because the control process exited with error code.
See "systemctl status yggdrasil.service" and "journalctl -xeu yggdrasil.service" for details.
root@thinkpad-p1:/home/jhnidek/github/redhat-insights/yggdrasil# systemctl status yggdrasil.service 
× yggdrasil.service - yggdrasil system service
     Loaded: loaded (/usr/lib/systemd/system/yggdrasil.service; disabled; preset: disabled)
    Drop-In: /usr/lib/systemd/system/service.d
             └─10-timeout-abort.conf
     Active: failed (Result: exit-code) since Mon 2024-05-06 14:07:11 CEST; 8s ago
       Docs: https://github.com/RedHatInsights/yggdrasil
    Process: 80880 ExecStart=/usr/bin/yggd (code=exited, status=1/FAILURE)
   Main PID: 80880 (code=exited, status=1/FAILURE)
        CPU: 153ms

kvě 06 14:07:11 thinkpad-p1 systemd[1]: Starting yggdrasil.service - yggdrasil system service...
kvě 06 14:07:11 thinkpad-p1 yggd[80880]: [yggd] 2024/05/06 14:07:11 /home/jhnidek/github/redhat-insights/yggdrasil/builddir/dist/srpm/yggdrasil-0.4.1/_build/src/github.com/redhatinsights/yggdrasil/cmd/yggd/m>
kvě 06 14:07:11 thinkpad-p1 yggd[80880]: [yggd] 2024/05/06 14:07:11 /home/jhnidek/github/redhat-insights/yggdrasil/builddir/dist/srpm/yggdrasil-0.4.1/_build/src/github.com/redhatinsights/yggdrasil/internal/w>
kvě 06 14:07:11 thinkpad-p1 yggd[80880]: cannot setup client: cannot connect client: cannot connect dispatcher: cannot request name on bus: Request to own name refused by policy
kvě 06 14:07:11 thinkpad-p1 systemd[1]: yggdrasil.service: Main process exited, code=exited, status=1/FAILURE
kvě 06 14:07:11 thinkpad-p1 systemd[1]: yggdrasil.service: Failed with result 'exit-code'.
kvě 06 14:07:11 thinkpad-p1 systemd[1]: Failed to start yggdrasil.service - yggdrasil system service.
```

It seems that some policy file is not enough... I will add comment to the code.

Reply:
The purpose of the group is to permit processes to send to the
`com.redhat.Yggdrasil1.Dispatcher1` destination. What is the advantage of making
workers use a separate group from the main process?

```xml
  <policy group="@group@">
    <!-- Only members of the @group@ group can send messages to Dispatcher1
    destination. -->
    <allow send_destination="com.redhat.Yggdrasil1.Dispatcher1" />
  </policy>
```